### ISSUE TITLE
L010: SQLFluff 0.9.1 (and later) capitalize columns named year/month when used as the second parameters in PARSE_DATE

### ISSUE THREAD

**Issue by kdw2126** [sqlfluff/sqlfluff#2508]
## Expected Behaviour
When run on a SQL file containing a line like PARSE_DATE("%Y", year), SQLFluff should realize that year must be a variable rather than an SQL keyword this context and not request that it is capitalized.

## Observed Behavior
For this function year and month is treated like SQL keywords rather than column names when used in the second argument.

## Steps to Reproduce
(1). Generate a SQL file of the form:
```sql
SELECT
    PARSE_DATE("%Y%m", CONCAT(year, month)) AS date
FROM
    model
```
(2). Set BigQuery as the dialect and run sqlfluff lint on this file.

Output is:
```
L:   1 | P:  34 | L010 | Keywords must be upper case.
L:   1 | P:  40 | L010 | Keywords must be upper case.
```

## Dialect
BigQuery

## Version
SQLFluff version is 0.9.3 (although this is not an issue in SQLFluff 0.9.0).  Python version is 3.8.1.0.
(I'm using DBT but the same issue occurs when not using dbt as a template).

## Configuration
```
[sqlfluff]
dialect = bigquery
exclude_rules = L003,L008,L011,L014,L016,L029,L031,L034

[sqlfluff:rules]
max_line_length = 120
comma_style = leading

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

**Comment by kdw2126** [comment]
Here's the output of _sqlfluff parse_:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'PARSE_DATE'
[L:  1, P: 18]      |                        bracketed:
[L:  1, P: 18]      |                            start_bracket:                    '('
[L:  1, P: 19]      |                            [META] indent:
[L:  1, P: 19]      |                            expression:
[L:  1, P: 19]      |                                literal:                      '"%Y%m"'
[L:  1, P: 25]      |                            comma:                            ','
[L:  1, P: 26]      |                            whitespace:                       ' '
[L:  1, P: 27]      |                            expression:
[L:  1, P: 27]      |                                function:
[L:  1, P: 27]      |                                    function_name:
[L:  1, P: 27]      |                                        function_name_identifier:  'CONCAT'
[L:  1, P: 33]      |                                    bracketed:
[L:  1, P: 33]      |                                        start_bracket:        '('
[L:  1, P: 34]      |                                        [META] indent:
[L:  1, P: 34]      |                                        date_part:            'year'
[L:  1, P: 38]      |                                        comma:                ','
[L:  1, P: 39]      |                                        whitespace:           ' '
[L:  1, P: 40]      |                                        date_part:            'month'
[L:  1, P: 45]      |                                        [META] dedent:
[L:  1, P: 45]      |                                        end_bracket:          ')'
[L:  1, P: 46]      |                            [META] dedent:
[L:  1, P: 46]      |                            end_bracket:                      ')'
[L:  1, P: 47]      |                    whitespace:                               ' '
[L:  1, P: 48]      |                    alias_expression:
[L:  1, P: 48]      |                        keyword:                              'AS'
[L:  1, P: 50]      |                        whitespace:                           ' '
[L:  1, P: 51]      |                        identifier:                           'date'
[L:  1, P: 55]      |            newline:                                          '\n'
[L:  2, P:  1]      |            [META] dedent:
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                from_expression:
[L:  3, P:  5]      |                    [META] indent:
[L:  3, P:  5]      |                    from_expression_element:
[L:  3, P:  5]      |                        table_expression:
[L:  3, P:  5]      |                            table_reference:
[L:  3, P:  5]      |                                identifier:                   'model'
[L:  3, P: 10]      |                    [META] dedent:
[L:  3, P: 10]      |    newline:                                                  '\n'
```

**Comment by tunetheweb** [comment]
ANSI assumes dateparts are identifiers UNLESS they are one of a set of date functions (it only lists `DATEADD`).
BigQuery assumes all dateparts are not identifiers.

We should add the list of datepart functions to BigQuery and reverse the assumption, so it's similar to ANSI.
